# Minecraft - DSCI 100 
###### Name:  Alycia M. 
###### Student Number: 92941665

In [1]:
# loading packages 
library(tidyverse)
library(tidymodels)
library(repr)
library(RColorBrewer)
library(patchwork)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

## Introduction 

Video games have evolved beyond entertainment into rich environments for behavioral research. As players interact within virtual worlds, they leave data that offer researchers opportunities to study human behavior and engagement patterns. 

Minecraft with millions of users worldwide, is particularly popular for such studies due to its open-ended gameplay and broad appeal across age groups.

At the University of British Columbia (UBC), a research team led by Dr. Frank Wood has set up a Minecraft server to collect data by 
recoridng players' actions as they navigate through the world all in which is to investigate how people play games and what factors influence their continued participation. However, running such a server is resource-intensive. It requires software licenses, robust hardware, and recruitment efforts to ensure enough players contribute meaningful data. To make this sustainable, the team is interested in identifying which types of players are most likely to stay engaged.

To aid in their work, will be answering the 
### Question: 
Can we predict whether a Minecraft player will subscribe based on their gender, age and hours played?




Analysis with be examined through player data set and session data set. 

## Analysis on Player Data Set

In [4]:
players <- read_csv("players.csv")
head(players,10)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1  Pro         TRUE    
2  Veteran     TRUE    
3  Veteran    FALSE    
4  Amateur     TRUE    
5  Regular     TRUE    
6  Amateur     TRUE    
7  Regular     TRUE    
8  Amateur    FALSE    
9  Amateur     TRUE    
10 Veteran     TRUE    
   hashedEmail                                                     
1  f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2  f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3  b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
4  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
5  7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
6  f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977
7  8e594b8953193b26f498db95a508b03c6fe1c24bb5251d392c18a0da9a722807
8  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
9  8b71f4d66a38389b7528bb38ba6eb71157733df7d1740371852a797ae97d82d1
10 bbe2d83de678f519c4b3daa7265e683b4fe2d814077f9094afd11d8f217039ec
   played_hours name      gender Age
1  30.3         Morgan    Male    9 
2   3.8         Christian Male   17 
3   0.0         Blake     Male   17 
4   0.7         Flora     Female 21 
5   0.1         Kylie     Male   21 
6   0.0         Adrian    Female 17 
7   0.0         Luna      Female 19 
8   0.0         Emerson   Male   21 
9   0.1         Natalie   Male   17 
10  0.0         Nyla      Female 22

In the player data set, it showcases  behavior and personal demographic information. It provides 7 variables in whch include : 

1. experience:
- in character
- showcases player skill level is a pro, veteran, amateur or regular.

2.  subscribe
- is logical
- indicates wheter or not the player is subscribed through true or false

3. hashedEmail
- in character
- An anonymized unique identifier for each player

4. played_hours
- in decimals
- Total hours the player has spent in the game

5. name
- in character
- Player's first name

6. gender
- in character
- player's gender

7. Age
- in decimals
- player's value

and there are 196 observations. 

## Analysis on Session Data Set

In [5]:
session <- read_csv("sessions.csv")
head(session,10)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8  ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9  96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
   start_time       end_time         original_start_time original_end_time
1  30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2  17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3  25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4  25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5  25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6  23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12      
7  15/04/2024 07:12 15/04/2024 07:21 1.71317e+12         1.71317e+12      
8  21/09/2024 02:13 21/09/2024 02:30 1.72688e+12         1.72689e+12      
9  21/06/2024 02:31 21/06/2024 02:49 1.71894e+12         1.71894e+12      
10 16/05/2024 05:13 16/05/2024 05:52 1.71584e+12         1.71584e+12

In the session data set, it captures detailed gameplay activity of an indivual. It contains 5 variables: 

1. hashedEmail
- in character
- An anonymized unique ID that links each session to a player

2. start_time
- in character
- 	Human-readable timestamp showing when the session began

3. end_time
-  in character
- Human-readable timestamp showing when the session ended

4. original_start_time
- in decimals
- timestamp when the session started

5. original_end_time
- in decimals
- timestamp when the session ended

and there are 1535 observations.

### Issues 

There are 2 data sets, in which would be beneficial to use both for the analysis. There are also more observation in the session data set compared to the player data set. Therefore, must find a way to combine the 2 data sets by hashed email and summarize session count and combine all required variables into one data set. 

Additionally, as gender is a character variable, need to adjust it so that it is suitable for the model and can be scaled to the other predictors as it can lead to an unbalance and unaccurate model.

----------------